From 2c092cafc24c93535ea47f1ee3a1a154a57ffc33 Mon Sep 17 00:00:00 2001
From: Ken Ma <make@marvell.com>
Date: Thu, 20 Jul 2017 14:41:13 +0800
Subject: [PATCH 1173/1345] ARM64: dts: marvell: add additional pinctrl
 definition for armada-37xx

commit  dbcbc6eb08b4409cbce6f7d86efa88a77fa0907b from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

Add mmc pins, pcie pins and sdio pins definition and do these pins'
configuration for DB board and espressobin board.

Cherry-picked from branch linux-4.12-devel and minor changes because of
the slightly different dts version.

Change-Id: I079aca83757591f538e2d127672f713f8b5fc884
Reviewed-on: http://vgitil04.il.marvell.com:8080/41890
Reviewed-by: Wilson Ding <dingwei@marvell.com>
Tested-by: Wilson Ding <dingwei@marvell.com>
Verified-Armada37x0: Wilson Ding <dingwei@marvell.com>
Signed-off-by: Ken Ma <make@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/42776
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Victor Gu <xigu@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/marvell/armada-3720-community.dts     |    4 ++-
 arch/arm64/boot/dts/marvell/armada-37xx.dtsi       |   26 +++++++++++---------
 2 files changed, 17 insertions(+), 13 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-3720-community.dts b/arch/arm64/boot/dts/marvell/armada-3720-community.dts
index bad9a5e..e6f5b72 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-community.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-community.dts
@@ -154,7 +154,7 @@
 				status = "okay";
 
 				pinctrl-names = "default";
-				pinctrl-0 = <&sdio_pins &cd_pins>;
+				pinctrl-0 = <&sdio_pins>;
 			};
 
 			sata@e0000 {
@@ -225,6 +225,8 @@
 };
 
 &pcie0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pcie_pins>;
 	status = "okay";
 	phys = <&a3700_comphy 1 COMPHY_PCIE0>;
 	phy-names = "comphy";
diff --git a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
index 16eef33..d45b7d0 100644
--- a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
@@ -260,6 +260,11 @@
 					groups = "uart2";
 					function = "uart";
 				};
+
+				mmc_pins: mmc-pins {
+					groups = "emmc_nb";
+					function = "emmc";
+				};
 			};
 
 			pinctrl_sb: pinctrl@18800 {
@@ -283,6 +288,15 @@
 					function = "mii";
 				};
 
+				sdio_pins: sdio-pins {
+					groups = "sdio_sb";
+					function = "sdio";
+				};
+
+				pcie_pins: pcie-pins {
+					groups = "pcie1";
+					function = "pcie";
+				};
 			};
 
 			eth0: ethernet@30000 {
@@ -340,13 +354,6 @@
 					marvell,pins = "mpp15", "mpp16";
 					marvell,function = "spi-quad";
 				};
-
-				mmc_pins: mmc-pins {
-					marvell,pins = "mpp27", "mpp28", "mpp29", "mpp30",
-							"mpp31", "mpp32", "mpp33", "mpp34", "mpp35";
-					marvell,function = "mmc";
-				};
-
 				uart1_pins: uart1_pins {
 					marvell,pins = "mpp18", "mpp19";
 					marvell,function = "uart2";
@@ -364,11 +371,6 @@
 				compatible = "marvell,armada-3700-sb-pinctrl";
 				reg = <0x18830 0x4>;
 
-				sdio_pins: sdio-pins {
-					marvell,pins = "mpp24", "mpp25", "mpp26", "mpp27", "mpp28", "mpp29";
-					marvell,function = "sdio";
-				};
-
 			};
 
 			u3d@50000 {
-- 
1.7.9.5

